llama-index-core/llama_index/core/workflow/decorators.py

```diff
@@ -30,6 +29,17 @@ def decorator(func: Callable) -> Callable:
         # This will raise providing a message with the specific validation failure
         validate_step_signature(func)
 
+        # Determine if this is a free function
+        name_toks = func.__qualname__.split(".")
+        is_free_func = len(name_toks) > 1 and name_toks[-2] == "<locals>"
```

Comment:
is this safe?

I tried really quickly in a script
```
def test():
    return True

print(test.__qualname__)
>>> "test"
```

`is_free_func` would be False in this case